src/common/sound.cpp

```diff
@@ -51,125 +79,42 @@ void Sound::stop() {
 // Every mode handle just his own signal types.
 void Sound::play(eSOUND_TYPE eSoundType) {
+    int t_size = 0;
     switch (eSoundMode) {
     case eSOUND_MODE_ONCE:
-        if (eSoundType == eSOUND_TYPE_Start) {
-            this->soundStart(1, 100.f);
-        }
-        if (eSoundType == eSOUND_TYPE_ButtonEcho) {
-            this->soundButtonEcho(1, 100.f);
-        }
-        if (eSoundType == eSOUND_TYPE_StandardPrompt) {
-            this->soundStandardPrompt(1, 500.f);
-        }
-        if (eSoundType == eSOUND_TYPE_StandardAlert) {
-            this->soundStandardAlert(1, 200.f);
-        }
-        if (eSoundType == eSOUND_TYPE_CriticalAlert) {
-            this->soundCriticalAlert(-1, 500.f);
-        }
-        break;
-    case eSOUND_MODE_LOUD:
-        if (eSoundType == eSOUND_TYPE_Start) {
-            this->soundStart(1, 100.f);
-        }
-        if (eSoundType == eSOUND_TYPE_ButtonEcho) {
-            this->soundButtonEcho(1, 100.0f);
-        }
-        if (eSoundType == eSOUND_TYPE_StandardPrompt) {
-            this->soundStandardPrompt(-1, 500.f);
-        }
-        if (eSoundType == eSOUND_TYPE_StandardAlert) {
-            this->soundStandardAlert(3, 200.f);
-        }
-        if (eSoundType == eSOUND_TYPE_CriticalAlert) {
-            this->soundCriticalAlert(-1, 500.f);
-        }
+        t_size = sizeof(onceTypes) / sizeof(onceTypes[0]);
+        this->_playSound(eSoundType, onceTypes, onceRepeats, t_size);
         break;
     case eSOUND_MODE_SILENT:
-        if (eSoundType == eSOUND_TYPE_Start) {
-            this->soundStart(1, 100.f);
-        }
-        if (eSoundType == eSOUND_TYPE_StandardAlert) {
-            this->soundStandardAlert(1, 200.f);
-        }
-        if (eSoundType == eSOUND_TYPE_CriticalAlert) {
-            this->soundCriticalAlert(-1, 500.f);
-        }
+        t_size = sizeof(silentTypes) / sizeof(silentTypes[0]);
+        this->_playSound(eSoundType, silentTypes, silentRepeats, t_size);
         break;
     case eSOUND_MODE_ASSIST:
-        if (eSoundType == eSOUND_TYPE_Start) {
-            this->soundStart(1, 100.f);
-        }
-        if (eSoundType == eSOUND_TYPE_ButtonEcho) {
-            this->soundButtonEcho(1, 100.0f);
-        }
-        if (eSoundType == eSOUND_TYPE_StandardPrompt) {
-            this->soundStandardPrompt(-1, 500.f);
-        }
-        if (eSoundType == eSOUND_TYPE_StandardAlert) {
-            this->soundStandardAlert(3, 200.f);
-        }
-        if (eSoundType == eSOUND_TYPE_EncoderMove) {
-            this->soundEncoderMove(1, 50.f);
-        }
-        if (eSoundType == eSOUND_TYPE_BlindAlert) {
-            this->soundBlindAlert(1, 100.f);
-        }
-        if (eSoundType == eSOUND_TYPE_CriticalAlert) {
-            this->soundCriticalAlert(-1, 500.f);
-        }
+        t_size = sizeof(assistTypes) / sizeof(assistTypes[0]);
+        this->_playSound(eSoundType, assistTypes, assistRepeats, t_size);
```

Comment:
why do you use explicit `this->` ?

Reply:
bad habit from previous languages - fixed